pkg/solver/matcher/matcher.go

```diff
@@ -56,40 +84,72 @@ func GetMatchingDeals(
 		// loop over resource offers
 		matchingResourceOffers := []data.ResourceOffer{}
 		for _, resourceOffer := range resourceOffers {
+			_, matchSpan := tracer.Start(ctx, "match",
+				trace.WithAttributes(attribute.String("job_offer.id", jobOffer.ID),
+					attribute.String("resource_offer.id", resourceOffer.ID)),
+			)
```

Comment:
Future work. Would like to refactor this part of the algorithm out into a function. The change would make reading this code easier and we could avoid defining an child span in the scope of its parent.